### PR TITLE
Checkout country select

### PR DESCRIPTION
**Pull request by ainarssondors**
Added select field for Checkout's step 1.
Requires My-Account-Country-Select

### CODE REVIEW

File: src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js [hunk 1]
```diff
@@ -144,8 +145,9 @@ class CheckoutShippingStep extends Component {
 
     renderField(id, overrideStateValue) {
         const { [id]: stateValue } = this.state;
+        const isSelect = id === 'country_id';
         const {
-            type = 'text',
+            type = isSelect ? 'select' : 'text',
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js]: Is this needed? You were able to overwrite type in fieldMap?

File: src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js [hunk 2]
```diff
@@ -144,8 +145,9 @@ class CheckoutShippingStep extends Component {
 
     renderField(id, overrideStateValue) {
         const { [id]: stateValue } = this.state;
+        const isSelect = id === 'country_id';
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js]: Please remove this.

File: src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js [hunk 3]
```diff
@@ -158,7 +160,7 @@ class CheckoutShippingStep extends Component {
               type={ type }
               label={ label }
               note={ note }
-              value={ overrideStateValue || stateValue }
+              value={ overrideStateValue || stateValue || (isSelect && DEFAULT_COUNTRY) }
```
Comment (alfredsgenkins) [inline src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js]: You can set override value or correct state value initially in this.state declaration

File: src/app/component/Field/Field.component.js [hunk 4]
```diff
@@ -332,7 +365,14 @@ Field.propTypes = {
         PropTypes.func,
         PropTypes.shape({ current: PropTypes.instanceOf(Element) })
     ]),
-    isAutocompleteAllowed: PropTypes.bool
+    isAutocompleteAllowed: PropTypes.bool,
+    countries: PropTypes.arrayOf(
```
Comment (alfredsgenkins) [inline src/app/component/Field/Field.component.js]: Why abstract select options are called countries?

File: src/app/component/Field/Field.component.js [hunk 5]
```diff
@@ -332,7 +365,14 @@ Field.propTypes = {
...
+        PropTypes.shape({
+            id: PropTypes.string,
+            label: PropTypes.string
+        })
+    ).isRequired,
+    getCountriesList: PropTypes.func.isRequired
```
Comment (alfredsgenkins) [inline src/app/component/Field/Field.component.js]: Why this one is here? This logic should be encapsulated somewhere, not in abstract field component.

File: src/app/component/Field/Field.container.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+/**
```
Comment (alfredsgenkins) [inline src/app/component/Field/Field.container.js]: Please get rid of this file.

File: src/app/component/Form/Form.component.js [hunk 7]
```diff
@@ -39,7 +40,7 @@ class Form extends Component {
             if (child && typeof child === 'object' && child.type && child.props) {
                 const { type: { name }, props, props: { children } } = child;
 
-                if (name === Field.prototype.constructor.name) {
+                if (name === Field.prototype.constructor.name || name === Select.prototype.constructor.name) {
```
Comment (alfredsgenkins) [inline src/app/component/Form/Form.component.js]: Is this needed? Seems like you moved select into Field.

File: src/app/component/Select/Select.component.js [hunk 8]
```diff
@@ -18,63 +18,91 @@ import './Select.style';
...
+                </li>
+            )
         ));
 
         return (
             <div block="Select" elem="Container">
                 <div block="Select" elem="Wrapper" tabIndex="0">
                     <div block="Select" elem="Current">
-                        <span>{ selectedFilter }</span>
+                        <span>{ selectedFilter || '' }</span>
```
Comment (alfredsgenkins) [inline src/app/component/Select/Select.component.js]: Is the a possibility to pass a placeholder? Let's not render empty string.

File: src/app/route/CheckoutPage/CheckoutPage.style.scss [hunk 9]
```diff
@@ -141,4 +141,12 @@
             margin-top: 0;
         }
     }
+
+    .Select-Container {
+            max-width: 40rem;
```
Comment (alfredsgenkins) [inline src/app/route/CheckoutPage/CheckoutPage.style.scss]: Wrong nesting here.

File: src/app/route/MyAccountDetails/MyAccountDetails.style.scss [hunk 10]
```diff
@@ -26,6 +30,10 @@
         margin-bottom: 1.9rem;
         border-right: 2px solid var(--color-breadcrumbs-background);
 
+        @include mobile {
+            display: none;
```
Comment (alfredsgenkins) [inline src/app/route/MyAccountDetails/MyAccountDetails.style.scss]: Let's alway hide sidebar for now (Antons request)

File: src/app/store/Field/Field.dispatcher.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import { QueryDispatcher, executePost } from 'Util/Request';
+import { prepareQuery } from 'Util/Query';
+import { getCountryList } from 'Store/Field';
+import FieldQuery from 'Query/FieldQuery.query';
+
+class FieldDispatcher extends QueryDispatcher {
```
Comment (alfredsgenkins) [inline src/app/store/Field/Field.dispatcher.js]: Maybe we should add this to HeaderAndFooter request? The first one on page?

File: src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js (hunk 2 above)
Comment (ainarssondors) [reply src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js]: Resolved.

File: src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js (hunk 1 above)
Comment (ainarssondors) [reply src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js]: Resolved.

File: src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js (hunk 3 above)
Comment (ainarssondors) [reply src/app/component/CheckoutShippingStep/CheckoutShippingStep.component.js]: Resolved.

File: src/app/component/Field/Field.component.js (hunk 4 above)
Comment (ainarssondors) [reply src/app/component/Field/Field.component.js]: Changed name to countryList

File: src/app/component/Field/Field.container.js (hunk 6 above)
Comment (ainarssondors) [reply src/app/component/Field/Field.container.js]: Resolved.

File: src/app/component/Form/Form.component.js (hunk 7 above)
Comment (ainarssondors) [reply src/app/component/Form/Form.component.js]: Resolved.

File: src/app/route/CheckoutPage/CheckoutPage.style.scss (hunk 9 above)
Comment (ainarssondors) [reply src/app/route/CheckoutPage/CheckoutPage.style.scss]: Moved max-width to select styling, all ul now has overflow: auto.

**Pull request merged**